tests/suites/os/tests/chrony/index.js

```diff
@@ -5,63 +5,43 @@ module.exports = {
 			title: 'Chronyd service',
 			run: async function(test) {
 				test.comment(`checking for chronyd service...`);
-				let result = '';
-				await this.context.get().utils.waitUntil(async () => {
-					result = await this.context
-						.get()
-						.worker.executeCommandInHostOS(
-							`systemctl is-active chronyd.service`,
-							this.context.get().link,
-						);
-					return result === 'active';
-				}, false);
-				result = await this.context
-					.get()
-					.worker.executeCommandInHostOS(
-						'systemctl status chronyd | grep running',
+				let p = this.context.get().utils.waitUntil(async () => {
+					return this.context.get().worker.executeCommandInHostOS(
+						`systemctl is-active chronyd.service`,
 						this.context.get().link,
-					);
-				test.is(result !== '', true, 'Chronyd service should be running');
+					).then((result) => {
+						return result === 'active';
+					});
+				}, false, 60, 1000);
+
+				return test.resolves(p, 'Chronyd service should be running');
 			},
 		},
 		{
 			title: 'Sync test',
 			run: async function(test) {
-				let result = '';
-				await this.context.get().utils.waitUntil(async () => {
+				let p = this.context.get().utils.waitUntil(async () => {
 					test.comment('checking system clock synchronized...');
-					result = await this.context
-						.get()
-						.worker.executeCommandInHostOS(
+					return this.context.get().worker.executeCommandInHostOS(
 							'timedatectl | grep System',
 							this.context.get().link,
-						);
-					return result === 'System clock synchronized: yes';
-				});
-				result = await this.context
-					.get()
-					.worker.executeCommandInHostOS(
-						'timedatectl | grep System',
-						this.context.get().link,
-					);
```

Comment:
@markcorbinuk Ditto here, we already waited above for the output of `timedatectl | grep System` to match `System clock synchronized: yes`, am I missing something?